operations/mimir/config.libsonnet

```diff
@@ -423,6 +425,13 @@
     service_ignored_labels:: [self.gossip_member_label],
   },
 
+  // Check configured deployment mode to ensure configuration is correct and consistent.
+  check_deployment_mode: if $._config.deployment_mode == 'microservices' || $._config.deployment_mode == 'read-write' then null else
+    error 'unsupported deployment mode "%s"' % $._config.deployment_mode,
+
+  check_deployment_mode_mutually_exclusive: if $._config.is_microservices_deployment_mode != $._config.is_read_write_deployment_mode then null else
+    error 'both is_microservices_deployment_mode and is_read_write_deployment_mode are set to the same value, while one and only one must be enabled',
```

Comment:
nit: the error message mentions the booleans that are set as a result of `deployment_mode`. Presumably `deployment_mode` is the thing that users should be changing?

Reply:
Actually, nevermind because I guess this can only happen if a user modifies the booleans themself which they shouldn't be doing.

Reply:
You're right. WDYT about https://github.com/grafana/mimir/pull/3475/commits/d59ee113e160f6b1bc7030039c06237423ff27d6?

Reply:
LGTM